docs/src/tutorial/cdef_classes.rst

```diff
@@ -43,25 +63,35 @@ little calling overhead compared to a cdef method.
 
 To make the class definitions visible to other modules, and thus allow for
 efficient C-level usage and inheritance outside of the module that
-implements them, we define them in a :file:`sin_of_square.pxd` file:
+implements them, we define them in:
 
 .. literalinclude:: ../../examples/tutorial/cdef_classes/sin_of_square.pxd
+    :caption: sin_of_square.pxd
 
 Using this, we can now change our integration example:
 
-.. literalinclude:: ../../examples/tutorial/cdef_classes/integrate.pyx
+.. tabs::
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/cdef_classes/integrate.py
+            :caption: integrate.py
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/cdef_classes/integrate.pyx
+             :caption: integrate.pyx
 
 This is almost as fast as the previous code, however it is much more flexible
 as the function to integrate can be changed. We can even pass in a new
 function defined in Python-space::
 
-  >>> import integrate
-  >>> class MyPolynomial(integrate.Function):
-  ...     def evaluate(self, x):
-  ...         return 2*x*x + 3*x - 10
-  ...
-  >>> integrate(MyPolynomial(), 0, 1, 10000)
-  -7.8335833300000077
+    >>> import integrate
+    >>> class MyPolynomial(integrate.Function):
+    ...     def evaluate(self, x):
+    ...         return 2*x*x + 3*x - 10
+    ...
+    >>> integrate(MyPolynomial(), 0, 1, 10000)
+    -7.8335833300000077
```

Comment:
There's a few of these big chunks where the only thing is changed is the indentation.

Is there good reason for this change? If there isn't then I think leave the indentation as it was (it makes it harder to spot other changes)

Reply:
A good reason to add indentation is to keep it the same throughout the documentation.

It's just that it's not very convenient for me if in one `rst` indent - 4 spaces, in the other - 2

Reply:
@da-woods yes, I agree that because of this it is more difficult to see the real changes, so I can make a separate PR that would fix the indentation for all `rst` documents.

Reply:
Yeah, let's not move around indentation as part of this change.

Reply:
@da-woods @scoder Why would we reintroduce this inconsistency for indentation?
![image](https://user-images.githubusercontent.com/52697657/123969381-d545c600-d9c0-11eb-98c2-a73a34765ba1.png)

Reply:
@da-woods @scoder do you have any arguments against this?
Or do you suggest fixing all the indentation for `rst` in a separate PR?
And even if so, why not now for this document?

Reply:
It just seems like a lot of lines are being changed for pretty little benefit (probably just because that's how your editor intents .rst) and then next person to change the file will then change it to how their editor does .rst, etc..

However, I've slightly run out of interest in arguing about indentation of documentation and it isn't ultimately my decision so I don't feel strongly about it.

Reply:
> and then next person to change the file will then change it to how their editor does .rst, etc..

Is there some kind of agreement on this? That is, for large codebases, there are usually conventions for formatting and this usually includes information about the amount of indentation, if this is not yet available for cython, then it is better to do this.

Is there some kind of agreement on this? That is, for large codebases, there are usually conventions for formatting and this usually includes information about the amount of indentation, if this has not yet been created for cython, then it is better to do this.
But as far as I remember, there is a codestyle check in the tests, so it just needs to be changed a little.

Reply:
> and then next person to change the file will then change it to how their editor does .rst, etc..

Is there some kind of agreement on this? That is, for large codebases, there are usually conventions for formatting and this usually includes information about the amount of indentation, if this is not yet available for cython, then it is better to do this.

But as far as I remember, there is a codestyle check in the tests, so it just needs to be changed a little.

Reply:
Ok @scoder, would it be better if I remove those indentations and make a PR that formats all the docs to be 4 spaces indented everywhere? (which is the case for most documents)

(it seems this is again a change for which the `git` can unfairly `blame` me =] )

Reply:
Well, if @scoder felt it was better to leave inconsistency, so [be it](https://github.com/cython/cython/pull/4232/commits/ef17f0c286861c08436af1cd0f959d7a6fe5a4c3)

Can this PR be merged now?